ingestr/src/kafka/model.py

```diff
@@ -0,0 +1,159 @@
+    ):
+        """
+        Read options from CLI parameters.
+        """
+        output_format = None
+        include_fields = []
+        select_field = None
+        if format:
+            output_format = format[0]
+        if include:
+            output_format = "flexible"
+            include_fields = list(map(str.strip, include[0].split(",")))
+        if select:
+            output_format = "flexible"
+            select_field = select[0]
+        return cls(
+            key_type=key_type and key_type[0] or None,
+            value_type=value_type and value_type[0] or None,
+            format=output_format,
+            include=include_fields,
+            select=select_field,
+        )
+
+
+@attrs.define
+class KafkaEvent:
+    """
+    Manage details of a typical Kafka event/message/record.
+
+    https://kafka.apache.org/intro#intro_concepts_and_terms
+    """
+
+    ts: Tuple[int, int]
+    topic: str
+    partition: int
+    offset: int
+    key: Union[bytes, str]
+    value: Union[bytes, str]
+
+    def decode_text(self):
+        """
+        Assume `key` and `value` are bytes but decode from UTF-8 well.
+        """
+        if self.key is not None and isinstance(self.key, bytes):
+            self.key = self.key.decode("utf-8")
+        if self.value is not None and isinstance(self.value, bytes):
+            self.value = self.value.decode("utf-8")
+
+    def to_dict(self, options: KafkaDecodingOptions):
+        """
+        Convert Kafka event to designated output format/layout.
+        """
+        # TODO: Make decoding from text optional.
+        self.decode_text()
+
+        message_id = digest128(self.topic + str(self.partition) + str(self.key))
+
+        # The standard message layout as defined per dlt and ingestr.
+        standard_payload = {
+            "partition": self.partition,
+            "topic": self.topic,
+            "key": self.key,
+            "offset": self.offset,
+            "ts": {
+                "type": self.ts[0],
+                "value": ensure_pendulum_datetime(self.ts[1] / 1e3),
+            },
+            "data": self.value,
+        }
+
+        # Basic Kafka message processors providing two formats.
+        # Returns the message value and metadata.
+        # The legacy format `standard_v1` uses the field `_kafka_msg_id`,
+        # while the future `standard_v2` format uses `_kafka__msg_id`,
+        # better aligned with all the other fields.
+        #
+        # Currently, as of July 2025, `standard_v1` is used as the
+        # default to not cause any breaking changes.
+
+        if options.format == "standard_v1":
+            UserWarning(
+                "Future versions of ingestr will use the `standard_v2` output format. "
+                "To retain compatibility, make sure to start using `format=standard_v1` early."
+            )
+            return {
+                "_kafka": standard_payload,
+                "_kafka_msg_id": message_id,
+            }
```

Comment:
This section might need further refinement.

a) Is the user warning emitted at all? Let's check that using a test case.
b) What if the user decides to deliberately continue using `standard_v1`? Can we have a mechanism to suppress the user warning in this case?